src/SFML/Audio/SoundBuffer.cpp

```diff
@@ -165,59 +181,56 @@ bool SoundBuffer::saveToFile(const std::filesystem::path& filename) const
 ////////////////////////////////////////////////////////////
 const std::int16_t* SoundBuffer::getSamples() const
 {
-    return m_samples.empty() ? nullptr : m_samples.data();
+    return m_impl->m_samples.empty() ? nullptr : m_impl->m_samples.data();
 }
 
 
 ////////////////////////////////////////////////////////////
 std::uint64_t SoundBuffer::getSampleCount() const
 {
-    return m_samples.size();
+    return m_impl->m_samples.size();
 }
 
 
 ////////////////////////////////////////////////////////////
 unsigned int SoundBuffer::getSampleRate() const
 {
-    ALint sampleRate;
-    alCheck(alGetBufferi(m_buffer, AL_FREQUENCY, &sampleRate));
-
-    return static_cast<unsigned int>(sampleRate);
+    return m_impl->m_sampleRate;
 }
 
 
 ////////////////////////////////////////////////////////////
 unsigned int SoundBuffer::getChannelCount() const
 {
-    ALint channelCount;
-    alCheck(alGetBufferi(m_buffer, AL_CHANNELS, &channelCount));
+    return m_impl->m_channelCount;
+}
+
 
-    return static_cast<unsigned int>(channelCount);
+////////////////////////////////////////////////////////////
+std::vector<SoundChannel> SoundBuffer::getChannelMap() const
+{
+    return m_impl->m_channelMap;
 }
 
 
 ////////////////////////////////////////////////////////////
 Time SoundBuffer::getDuration() const
 {
-    return m_duration;
+    return m_impl->m_duration;
 }
 
 
 ////////////////////////////////////////////////////////////
 SoundBuffer& SoundBuffer::operator=(const SoundBuffer& right)
 {
-    SoundBuffer temp(right);
-
-    std::swap(m_samples, temp.m_samples);
-    std::swap(m_buffer, temp.m_buffer);
-    std::swap(m_duration, temp.m_duration);
+    const SoundBuffer& temp(right);
```

Comment:
Is this supposed to be a reference?

Reply:
clang-tidy warned about this not being a `const&` so I added that but it did confuse me too. Swapping out of `const&` doesn't make sense to me.

Reply:
I think it's actually correct. The thing being swapped is the pointed-to data owned by `temp.m_impl`. `temp` and `m_impl` don't actually change.

Reply:
Why even have the temp variable at all then? And I guess it would mean the comment about temp being destructed doesn't make sense?

Reply:
This is indeed moving from temp.m_impl but temp is the same instance as right, so this is moving from right.m_impl, which is not correct for a copy constructor.